includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -957,10 +935,19 @@ public function process_payment_for_order( $cart, $payment_information, $additio
 		$amount   = $order->get_total();
 		$metadata = $this->get_metadata_from_order( $order, $payment_information->get_payment_type() );
 
-		$customer_details_options   = [
-			'is_woopay' => filter_var( $metadata['paid_on_woopay'] ?? false, FILTER_VALIDATE_BOOLEAN ),
-		];
-		list( $user, $customer_id ) = $this->manage_customer_details_for_order( $order, $customer_details_options );
+		$user = $this->get_user_for_order( $order );
+
+		// Determine the customer making the payment, create one if we don't have one already.
+		$customer_id = $this->customer_service->get_customer_id_by_user_id( $user->ID );
+		if ( null === $customer_id ) {
+			$customer_id = $this->create_customer_using_order( $order, $user );
+		} else {
+			// No need to update the customer object if we just created it.
+			$customer_details_options = [
+				'is_woopay' => filter_var( $metadata['paid_on_woopay'] ?? false, FILTER_VALIDATE_BOOLEAN ),
+			];
+			$this->update_customer_with_order_data_on_shutdown( $customer_id, $order, $customer_details_options );
+		}
```

Comment:
This specific if-statement might be better as a `create_or_update_customer` function?

Reply:
Created and used the helper function in d27692549b43ccb509771c22eda34bde1175655d, f2186e12e4201f46fcf165e43b2459c6c126b37a, and f2b19a3b06612b7225c3e6ad460f27d8fe205696.